drivers/dma/dma_mcux_lpc.c

```diff
@@ -188,10 +190,43 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 
 	LOG_DBG("channel is %d", p_handle->channel);
 
-	if (config->source_chaining_en && config->dest_chaining_en) {
-		LOG_DBG("link dma out 0 to channel %d", config->linked_channel);
-		/* Link DMA_OTRIG 0 to channel */
-		INPUTMUX_AttachSignal(INPUTMUX, 0, config->linked_channel);
+	if (config->source_chaining_en || config->dest_chaining_en) {
+		if (!k_sem_take(&is_trig_mux_in_use, K_MSEC(200))) {
+			LOG_DBG("link dma out 0 to channel %d", config->linked_channel);
+			/* At the moment we only have support
+			 * for OTrig and are planning to add
+			 * support for X amount of OTrigs that
+			 * each SOC supports that use DMA_MCUX_LPC
+			 */
+
+			/* Since INPUTMUX handles the signal for
+			 * channel chaining, we should request that
+			 * the trigger is handled via INPUTMUX hardware
+			 * trigger by default
+			 */
+			DEV_BASE(dev)->CHANNEL[config->linked_channel].CFG |=
+				DMA_CHANNEL_CFG_HWTRIGEN_MASK;
+			DMA_EnableChannel(DEV_BASE(dev), config->linked_channel);
+#if CONFIG_SOC_LPC55S69_CPU0
+			/* Link DMA_OTRIG 0 to channel */
+			INPUTMUX_AttachSignal(INPUTMUX, 0,
+				kINPUTMUX_Dma0Hash0TxTrigoutToTriginChannels
+				+ channel);
+			INPUTMUX_AttachSignal(INPUTMUX, config->linked_channel,
+					kINPUTMUX_Otrig0ToDma0);
+#else
+			k_sem_give(&is_trig_mux_in_use);
+			LOG_ERR("\nAt this moment, this driver does not support"
+			"\n channel chaining for this SOC -> %s\n", CONFIG_SOC);
+			return -ENOTSUP;
+#endif /* CONFIG_SOC_LPC55S69_CPU0 */
+			k_sem_give(&is_trig_mux_in_use);
```

Comment:
again, giving the semaphore here defeats its purpose, you need to remove this

Reply:
if you want to give the semaphore here, you have to change the paradigm and track whether or not a channel is using the mux in another variable rather than the semaphore being taken inherently meaning that